include/alpaka/atomic/AtomicStlLock.hpp

```diff
@@ -70,9 +70,12 @@ namespace alpaka
             ALPAKA_FN_ACC_NO_CUDA /*virtual*/ ~AtomicStlLock() = default;
 
         private:
-            std::mutex mutable m_mtxAtomic; //!< The mutex protecting access for a atomic operation.
+            static std::mutex m_mtxAtomic; //!< The mutex protecting access for a atomic operation.
         };
 
+        // initialize the static member of the class AtomicStlLock
+        std::mutex AtomicStlLock::m_mtxAtomic;
```

Comment:
This does not compile.
You might have a look at [this Stackoverflow question](http://stackoverflow.com/questions/11709859/how-to-have-static-data-members-in-a-header-only-library).